ibis/expr/sql.py

```diff
@@ -371,18 +371,17 @@ def to_sql(
             # default to duckdb for SQL compilation because it supports the
             # widest array of ibis features for SQL backends
             backend = ibis.duckdb
-            read = "duckdb"
-            write = ibis.options.sql.default_dialect
+            dialect = ibis.options.sql.default_dialect
         else:
-            read = write = backend.dialect
+            dialect = backend.dialect
     else:
         try:
             backend = getattr(ibis, dialect)
         except AttributeError:
             raise ValueError(f"Unknown dialect {dialect}")
         else:
-            read = write = getattr(backend, "dialect", dialect)
+            dialect = getattr(backend, "dialect", dialect)
 
-    sql = backend._to_sql(expr.unbind(), **kwargs)
-    (transpiled,) = sg.transpile(sql, read=read, write=write, pretty=pretty)
-    return SQLString(transpiled)
+    sg_expr = backend._to_sqlglot(expr.unbind(), **kwargs)
```

Comment:
I think this should rather be `backend.compile(expr.unbind(), **kwargs)`

Reply:
And the backend could be chosen given the `dialect`.

Reply:
After a second thought, changing it would align with the ibis backends, but if we want to support all dialects in sqlglot then it shouldn't be changed.

Reply:
I think we can visit changing it in a follow-up if we do it at all. I'm inclined to leave compile returning strings for now.